app/helpers/layout_helper.rb

```diff
@@ -347,6 +347,18 @@ def fullscreen_button(element = "$(this).prev()")
   private
 
   def authorized_associations(associations)
-    associations.included_modules.include?(Authorizable) ? associations.authorized : associations
+    if associations.included_modules.include?(Authorizable)
+      if associations.respond_to?(:klass)
+        associations.authorized(authorized_assocations_permission_name(associations.klass), associations.klass)
+      else
+        associations.authorized(authorized_assocations_permission_name(associations), associations)
+      end
+    else
+      associations
+    end
+  end
+
+  def authorized_assocations_permission_name(klass)
+    "view_#{klass.to_s.downcase.pluralize}"
```

Comment:
> What do you want me to test? Maybe that authorized_associations method can handle situation when there's no permission for associated the resource? What would be the right behaviour in this case?

I'm most concerned that we should test that this method resolves the correct permission for anything in Foreman that would be shown on this form (Taxonomix stuff), to prevent regressions where a non-admin user can't see resources with any combination of permissions.

Perhaps something for a unit or functional test, with an org that's associated to one of everything and a non-user, to show that the method shows every resource?  It doesn't help as much with plugins, but ensures core is OK.

> Currently, authorized scope wouldn't limit on specific permission but take resources that are granted via any permission (like before this patch). Maybe such resource shouldn't include Authorizable module in the first place...

Seems a reasonable behaviour, OK.  So `Foo.authorized("unknown")` would return resources for which the user has `edit_foo` for instance?